web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx

```diff
@@ -80,15 +186,87 @@ export function SetupConnect(props: AgentStepProps) {
         </ButtonSecondary>
       </StyledBox>
 
+      {hint}
+
       <ActionButtons
-        onProceed={() => {}}
-        disableProceed={true}
+        onProceed={props.nextStep}
+        disableProceed={!node}
         onPrev={props.prevStep}
       />
     </Flex>
   );
 }
 
+/**
+ * usePollForConnectMyComputerNode polls for a Connect My Computer node that joined the cluster
+ * after starting opening the SetupConnect step.
+ *
+ * The first polling request fills out a set of node IDs (initialNodeIdsRef). Subsequent requests
+ * check the returned nodes against this set. The hook stops polling as soon as a node that is not
+ * in the set was found.
+ *
+ * Unlike the DownloadScript step responsible for adding a server, we don't have a unique ID that
+ * identifies the node that the user added after following the steps from the guided flow. In
+ * theory, we could make the deep link button pass such ID to Connect, but the user would still be
+ * able to just launch the app directly and not use the link.
+ *
+ * Because of that, we must depend on comparing the list of nodes against the initial set of IDs.
+ */
+export const usePollForConnectMyComputerNode = (args: {
```

Comment:
Should we have any timeout? From what I see, `useCreateDatabase` waits for 30 seconds.
We are waiting indefinitely, maybe we could limit it to say 20 minutes?

 OTOH I don't see a timeout for other places that use `usePoll`, so perhaps we are fine with it.

Reply:
Right, other places in Discover which do polling AFAIK don't have any timeout so I followed suit. I assume that the Discover team talked about this and this was deemed to be the best course of action.

Reply:
i'm sorry.. i meant WITHOUT timeout 😅